### DISCUSSION TITLE
List installed modules and versions

### DISCUSSION THREAD

**Discussion by arossert** [pyinstaller/pyinstaller#6120] [PyInstaller]
Is there a way to list all collected modules and their versions from within the compiled file?

I have tied multiple solution that works outside of PyInstaller but all failed
`pkg_resources`, `importlib_metadata`, `pip._internal.utils.misc.get_installed_distributions`, `pipdeptree`

Looks like it is a limitation of PyInstaller, if this is not possible, can I at least get it from the build process? (even by parsing some log files)

**Comment by rokm** [comment]
PyInstaller does not collect metadata for packages, unless requested by hooks (or if bytecode scanner detects the use of certain functions that require it).

But you could reconstruct the list of collected packages by parsing `build/<name>/Analysis-00.toc`.

**Reply by arossert** [reply]
Thanks, @rokm for the info, I looked at the `Analysis-00.toc` file and I could not find a way to extract the collected modules, it only has the collected files.

**Reply by rokm** [reply]
The collected modules are also files (with `PYMODULE` type for pure-python modules)...
